### PR TITLE
Consolidate HTTP exceptions

### PR DESCRIPTION
**Pull request by shadowhand**
Having one class for HTTP exceptions is more predictable and can be
handled easier.

With a single class it is no longer necessary to use method checking
to apply additional modifications to the response.

By using a HTTP status code validator it is possible to provide the
HTTP status code as the exception code, without a separate method.

Refs #75

### CODE REVIEW

File: src/Exception/HttpException.php [hunk 1]
```diff
@@ -1,28 +1,42 @@
...
-    public static function invalidAccessMethod($method, $path, array $allowed)
+    public static function methodNotAllowed($method, $path, array $allowed)
     {
         $error = new static(sprintf(
             'Cannot access resource `%s` using method `%s`',
             $path,
             $method,
             implode(', ', $allowed)
-        ));
+        ), 405);
```
Comment (elazar) [inline src/Exception/HttpException.php]: Doesn't look like the list of allowed methods is actually included in the error or exception instance.

File: src/Exception/HttpException.php [hunk 2]
```diff
@@ -1,28 +1,42 @@
...
      * @return static
      */
-    public static function invalidAccessMethod($method, $path, array $allowed)
+    public static function methodNotAllowed($method, $path, array $allowed)
     {
         $error = new static(sprintf(
             'Cannot access resource `%s` using method `%s`',
             $path,
             $method,
             implode(', ', $allowed)
```
Comment (elazar) [inline src/Exception/HttpException.php]: Doesn't look like the list of allowed methods is actually included in the exception message.

Comment (shadowhand) [inline src/Exception/HttpException.php]: Fixed.

**Comment by elazar** [comment]
:+1:

**Pull request merged**